### PR TITLE
FEATURE: change layout when default page is category to tabular for _…

### PR DESCRIPTION
**Pull request by mrfinch**
…escaped_fragment_ page

This is a part of https://meta.discourse.org/t/improving-discourse-static-html-archive/112497

The old layout can be viewed at https://meta.discourse.org/categories/?_escaped_fragment_

I have removed `itemprop='item'` since it was not needed (all information is provided in ListItem already) and was showing up as an error here: https://search.google.com/structured-data/testing-tool/u/0/#url=https%3A%2F%2Fmeta.discourse.org%2Fcategories%2F%3F_escaped_fragment_

Attaching screenshot with the new layout.

Screenshot from 2019-03-27 22-24-51

### CODE REVIEW

File: app/assets/stylesheets/desktop/category-list.scss [hunk 1]
```diff
@@ -109,6 +109,14 @@
     .latest {
       padding: 0 0 10px 10px;
     }
+
+    .blue-border {
+      border-color: #0088cc;
```
Comment (ZogStriP) [inline app/assets/stylesheets/desktop/category-list.scss]: Here, we should be using the colors in the scss variables. That way, if the admins change their theme, the colors will change as well.

Comment (mrfinch) [reply app/assets/stylesheets/desktop/category-list.scss]: i have fixed it now @ZogStriP

File: app/views/categories/index.html.erb [hunk 2]
```diff
@@ -1,17 +1,34 @@
...
+  <table class='category-list'>
+    <thead>
+      <tr>
+        <th class='category'><%= t 'js.categories.category' %></th>
+        <th class='topics'><%= t 'js.topic.list' %></th>
+      </tr>
+    </thead>
+    <tbody>
+      <meta itemprop='itemListOrder' content='http://schema.org/ItemListOrderDescending'>
+      <% @category_list.categories.each_with_index do |c, index| %>
```
Comment (ZogStriP) [inline app/views/categories/index.html.erb]: You don't need to keep track of the index just to set an odd/even class. You can do that using CSS only 😉 

https://www.w3.org/Style/Examples/007/evenodd.en.html

Comment (mrfinch) [reply app/views/categories/index.html.erb]: Haha, actually I didn't add an index in the loop. It's already used in this line: ` <meta itemprop='position' content='<%= index %>'>` , we need it for SEO @ZogStriP

Comment (ZogStriP) [reply app/views/categories/index.html.erb]: My bad, I missed it. You should still use CSS to do the odd/even class ;)

Comment (mrfinch) [reply app/views/categories/index.html.erb]: Haha, I have made the change @ZogStriP

Comment (ZogStriP) [reply app/views/categories/index.html.erb]: Sorry, I just looked at your screenshot and I wonder: why aren't we using the category color for the border?

Comment (mrfinch) [reply app/views/categories/index.html.erb]: Actually, I didn't know that the category had colour stored. I have fixed it now, thanks @ZogStriP  :bowing_man:

File: app/views/categories/index.html.erb [hunk 3]
```diff
@@ -1,17 +1,34 @@
...
+        <tr>
+          <td class='category' style='border-color: #<%= c.color %>;'>
+            <div itemprop='itemListElement' itemscope itemtype='http://schema.org/ListItem'>
+              <meta itemprop='position' content='<%= index %>'>
+              <meta itemprop='url' content='<%= "#{c.url}" %>'>
```
Comment (ZogStriP) [inline app/views/categories/index.html.erb]:
```suggestion
              <meta itemprop='url' content='<%= c.url %>'>
```

File: app/views/categories/index.html.erb [hunk 4]
```diff
@@ -1,17 +1,34 @@
...
+              <h3>
+                <a href='<%= "#{c.url}" %>'>
```
Comment (ZogStriP) [inline app/views/categories/index.html.erb]:
```suggestion
                <a href='<%= c.url %>'>
```

File: app/views/categories/index.html.erb [hunk 5]
```diff
@@ -1,17 +1,34 @@
...
+                  <span itemprop='name'><%= c.name %></span>
+                </a>
+              </h3>
+              <div itemprop='description'><%= c.description&.html_safe %></div>
+            </div>
+          </td>
+          <td class='topics'>
+            <div title='<%= "#{c.topic_count}" %> <%= t 'js.topic.list' %>'><%= c.topic_count %></div>
```
Comment (ZogStriP) [inline app/views/categories/index.html.erb]:
```suggestion
            <div title='<%= c.topic_count %> <%= t 'js.topic.list' %>'><%= c.topic_count %></div>
```

File: app/views/categories/index.html.erb (hunk 3 above)
Comment (mrfinch) [reply app/views/categories/index.html.erb]: Actually, it's an old code. I had just changed the indentation :sweat_smile:, I have fixed all your suggestions @ZogStriP

**Comment by ZogStriP** [comment]
Thanks 👍

**Pull request merged**